athena-msk/src/main/java/com/amazonaws/athena/connectors/msk/serde/MskJsonDeserializer.java

```diff
@@ -0,0 +1,100 @@
+ * 
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+package com.amazonaws.athena.connectors.msk.serde;
+
+import com.amazonaws.athena.connectors.msk.AmazonMskUtils;
+import com.amazonaws.athena.connectors.msk.dto.MSKField;
+import com.amazonaws.athena.connectors.msk.dto.Message;
+import com.amazonaws.athena.connectors.msk.dto.TopicResultSet;
+import com.fasterxml.jackson.databind.JsonNode;
+import org.apache.arrow.vector.types.pojo.Schema;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.nio.charset.StandardCharsets;
+
+public class MskJsonDeserializer extends MskDeserializer
+{
+    private static final Logger LOGGER = LoggerFactory.getLogger(MskJsonDeserializer.class);
+    public MskJsonDeserializer(Schema schema)
+    {
+        super(schema);
+    }
+
+    @Override
+    public TopicResultSet deserialize(String topic, byte[] data)
+    {
+        if (data == null) {
+            return null;
+        }
+
+        // Initiating TopicResultSet pojo to put the raw data.
+        TopicResultSet topicResultSet = new TopicResultSet();
+        topicResultSet.setTopicName(topic);
+        topicResultSet.setDataFormat(Message.DATA_FORMAT_JSON);
+
+        try {
+            // Transforming the topic raw (json) data to JsonNode using ObjectMapper.
+            JsonNode json = AmazonMskUtils
+                    .getObjectMapper()
+                    .readValue(new String(data, StandardCharsets.UTF_8), JsonNode.class);
+
+            // Creating Field object for each fields in raw data.
+            // Also putting additional information in fields from fields metadata.
+            schema.getFields().forEach(field -> {
+                String mapping = field.getMetadata().get("mapping");
+                try {
+                    topicResultSet.getFields().add(new MSKField(
+                            field.getName(),
+                            mapping,
+                            field.getMetadata().get("type"),
+                            field.getMetadata().get("formatHint"),
+                            castValue(field, json.get(mapping).asText())
+                    ));
+                }
+                catch (Exception e) {
+                    LOGGER.error("MskJsonDeserializer: Error in castValue : while converting raw value to typed value");
+                    e.printStackTrace();
+                }
+            });
+            return topicResultSet;
+        }
+        catch (Exception e) {
+            LOGGER.error("MskJsonDeserializer: Error when deserializing byte[] to TopicResultSet");
+            e.printStackTrace();
```

Comment:
nit:

Use the version of LOGGER.error where you can include the exception as the last parameter. This will automatically log the stack trace for you: https://logging.apache.org/log4j/2.x/log4j-api/apidocs/org/apache/logging/log4j/Logger.html#error-java.lang.String-java.lang.Throwable-

I'm not sure if `e.printStackTrace()` will actually show up in our logs since that will be going to stdout instead of the logger.

Make sure to address this in all other locations that you do `e.printStackTrace()` as well.